src/volume/volume_gaussian.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+	Lightmetrica - Copyright (c) 2019 Hisanari Otsu
+	Distributed under MIT license. See LICENSE file for details.
+*/
+
+#include <pch.h>
+#include <lm/core.h>
+#include <lm/volume.h>
+
+LM_NAMESPACE_BEGIN(LM_NAMESPACE)
+
+/*
+\rst
+.. function:: volume::gaussian
+
+    Gaussian volume.
+
+    :param color color: Stored color value.
+    :param float scalar: Stored max scalar value.
+    :param Vec3 pos: Center position of the volume
+    :param Vec3 sigma: Gaussian parameter of standard deviation for x,y,z axis
+\endrst
+*/
+class Volume_Gaussian : public Volume {
+private:
+	Bound bound_;// Computed from 0.001% of the max of gaussian, then times two
+	std::optional<Vec3> color_;
+	std::optional<Float> scalar_;
+	Vec3 pos_;
+	Vec3 sigma_;
+
+public:
+	LM_SERIALIZE_IMPL(ar) {
+		ar(bound_);
+	}
+
+public:
+	virtual void construct(const Json& prop) override {
+		color_ = json::value_or_none<Vec3>(prop, "color");
+		scalar_ = json::value_or_none<Float>(prop, "scalar");
+		pos_ = json::value<Vec3>(prop, "pos", Vec3(.0_f, .0_f, .0_f));
+		sigma_ = json::value<Vec3>(prop, "sigma", Vec3(1._f, 1._f, 1._f));
+
+		if ( !color_ && !scalar_ ) {
+			LM_THROW_EXCEPTION(Error::InvalidArgument,
+				"Either 'color' or 'scalar' property is necessary.");
+		}
+
+		bound_ = computeBound(sigma_);
+	}
+
+    Bound computeBound(const Vec3& sigma) const {
```

Comment:
If this function is used only once, you can merge it to construct() function. I prefer to use lambda function inside the function to limit the scope, though I don't think it is simple enough to write it directly.